### PR TITLE
feat(vector): add API to append arc

### PR DESCRIPTION
**Pull request by XuNeo**
### Description of the feature or fix

The code originally comes from lv_vg_lite_path.c which refers to ThorVG implementation.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by XuNeo** [comment]
Hi @onecoolx ,
Please help to review the code.

**Review by C47D — APPROVED** [review]

File: src/draw/lv_draw_vector.c [hunk 1]
```diff
@@ -14,6 +14,23 @@
 #include "../stdlib/lv_string.h"
 #include <stdbool.h>
 #include <math.h>
+#include <float.h>
+
+#define MATH_PI  3.14159265358979323846f
+#define MATH_HALF_PI 1.57079632679489661923f
+#define MATH_TWO_PI 6.28318530717958647692f
+#define DEG_TO_RAD 0.017453292519943295769236907684886f
+#define RAD_TO_DEG 57.295779513082320876798154814105f
```
Comment (W-Mai) [inline src/draw/lv_draw_vector.c]: Is this necessary?

Comment (XuNeo) [reply src/draw/lv_draw_vector.c]: It's used be the newly added API, following the original code style.

File: src/draw/lv_draw_vector.c [hunk 2]
```diff
@@ -14,6 +14,23 @@
...
+
+#define MATH_TANF(x) tanf(x)
+#define MATH_SINF(x) sinf(x)
+#define MATH_COSF(x) cosf(x)
+#define MATH_ASINF(x) asinf(x)
+#define MATH_FABSF(x) fabsf(x)
+#define MATH_SQRTF(x) sqrtf(x)
```
Comment (kisvegabor) [inline src/draw/lv_draw_vector.c]: Do we need such trivial defines?

**Review by C47D — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Review by W-Mai — APPROVED** [review]

**Pull request merged**